test_vm/tests/batch_onboarding.rs

```diff
@@ -50,11 +51,30 @@ impl Onboarding {
 fn batch_onboarding(v2: bool) {
     let store = MemoryBlockstore::new();
     let v = TestVM::<MemoryBlockstore>::new_with_singletons(&store);
-    let addrs = create_accounts(&v, 1, &TokenAmount::from_whole(10_000));
     let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+
+    let proven_count = batch_onboarding_test(&v, v2, &seal_proof);
+
+    let network_stats = v.get_network_stats();
+    let sector_size = seal_proof.sector_size().unwrap() as u64;
+    assert_eq!(network_stats.total_bytes_committed, BigInt::from(sector_size * proven_count));
+    assert!(network_stats.total_pledge_collateral.is_positive());
+
+    v.expect_state_invariants(
```

Comment:
I think `check_state_invariants` and this method can similarly be implemented as helper functions in terms of the blockstore, and this can then be pushed down into the test. It'll still execute in Rust/native, should be no slower.